common/environment/environment.cpp

```diff
@@ -2179,6 +2307,51 @@ unsigned CConstDropZoneInfoIterator::count() const
     return maxIndex;
 }
 
+//--------------------------------------------------
+
+CConstDfuQueueInfoIterator::CConstDfuQueueInfoIterator()
+{
+    Owned<IEnvironmentFactory> factory = getEnvironmentFactory(true);
+    constEnv.setown((CLocalEnvironment *)factory->openEnvironment());
+    maxIndex = constEnv->getNumberOfDfuQueues();
+}
+
+bool CConstDfuQueueInfoIterator::first()
+{
+    index = 1;
+    curr.setown(constEnv->getDfuQueueByIndex(index));
+    return curr != nullptr;
+}
+
+bool CConstDfuQueueInfoIterator::next()
+{
+    if (index < maxIndex)
+    {
+        index++;
+        curr.setown(constEnv->getDfuQueueByIndex(index));
```

Comment:
FUTURE:
there's slight issue to this and all the other similar iterators..
If the environment updates whilst iterating, then the results will be indeterminate.

Not something to tackle now, but I suspect the way this should work, is to gather them into lists, once (e.g. on 1st request) - then the contents of the list at iterator construction time, so underlying list can be rebuild if env. changes without effecting the iterator.

It would also be more efficient this way, as generally doesn't change often and it wouldn't have to be recreating an environment factory/openEnvironment etc., every time an interator was required.

Reply:
@jakesmith can you open a JIRA for this?

Reply:
Have opened: https://track.hpccsystems.com/browse/HPCC-22177